File: android/app/src/main/java/com/interview/tracker/domain/model/Candidate.kt
package com.interview.tracker.domain.model

data class PhoneNumber(
    val number: String = "",
    val type: String = "primary",
    val isPrimary: Boolean = false
)

data class EmailAddress(
    val address: String = "",
    val type: String = "primary",
    val isPrimary: Boolean = false
)

data class Address(
    val street: String = "",
    val city: String = "",
    val state: String = "",
    val pincode: String = ""
)

data class EducationItem(
    val qualification: String = "",
    val course: String = "",
    val specialization: String = "",
    val passingYear: String = "",
    val institute: String = "",
    val percentageOrCgpa: String = ""
)

data class ExperienceItem(
    val company: String = "",
    val designation: String = "",
    val fromYear: String = "",
    val toYear: String = "",
    val roleSummary: String = "",
    val isCurrent: Boolean = false
)

/**
 * Domain model for a candidate, decoupled from API DTO.
 */
data class Candidate(
    val interviewId: Int = 0,
    val interviewIdString: String = "",
    val interviewDate: String = "",
    val candidateName: String = "",
    val mobileNo: String = "",
    val phones: List<PhoneNumber> = emptyList(),
    val emails: List<EmailAddress> = emptyList(),
    val dob: String = "",
    val age: Int? = null,
    val currentAddress: Address = Address(),
    val permanentAddress: Address = Address(),
    val isPermanentSameAsCurrent: Boolean = false,
    val educationList: List<EducationItem> = emptyList(),
    val education: String = "",
    val experienceList: List<ExperienceItem> = emptyList(),
    val totalExperience: String = "",
    val currentCompany: String = "",
    val currentDesignation: String = "",
    val currentLocation: String = "",
    val currentSalary: String = "",
    val expectedSalary: String = "",
    val noticePeriod: String = "",
    val positionApplied: String = "",
    val department: String = "",
    val skills: List<String> = emptyList(),
    val joiningAvailability: String = "",
    val technicalKnowledge: String = "",
    val recommendation: String = "",
    val finalStatus: String = "",
    val joiningDate: String = "",
    val interviewer: String = "",
    val resumeUrl: String = "",
    val resumeDocumentId: String = "",
    val whatsappOptIn: Boolean = true,
    val remarks: String = ""
) {
    val displayId: String
        get() = interviewIdString.ifBlank { if (interviewId > 0) "#$interviewId" else "Pending" }

    val allPhones: List<PhoneNumber>
        get() = if (phones.isNotEmpty()) phones else if (mobileNo.isNotBlank()) listOf(PhoneNumber(number = mobileNo, type = "primary", isPrimary = true)) else emptyList()
}

File: windows/InterviewTracker/InterviewTracker/Models/Candidate.cs
using System.Text.Json.Serialization;

namespace InterviewTracker.Models;

public class Candidate
{
    [JsonPropertyName("Interview ID")]
    public string InterviewId { get; set; } = "";

    [JsonPropertyName("Interview Date")]
    public string InterviewDate { get; set; } = "";

    [JsonPropertyName("Candidate Name")]
    public string CandidateName { get; set; } = "";

    [JsonPropertyName("Mobile No.")]
    public string MobileNo { get; set; } = "";

    [JsonPropertyName("Position Applied For")]
    public string PositionAppliedFor { get; set; } = "";

    [JsonPropertyName("Department")]
    public string Department { get; set; } = "";

    [JsonPropertyName("Education / Qualification")]
    public string EducationQualification { get; set; } = "";

    [JsonPropertyName("Total Experience (Years)")]
    public string TotalExperienceYears { get; set; } = "";

    [JsonPropertyName("Current Location")]
    public string CurrentLocation { get; set; } = "";

    [JsonPropertyName("Current Salary")]
    public string CurrentSalary { get; set; } = "";

    [JsonPropertyName("Expected Salary")]
    public string ExpectedSalary { get; set; } = "";

    [JsonPropertyName("Notice Period")]
    public string NoticePeriod { get; set; } = "";

    [JsonPropertyName("Joining Availability")]
    public string JoiningAvailability { get; set; } = "";

    [JsonPropertyName("Technical Knowledge (10)")]
    public string TechnicalKnowledge { get; set; } = "";

    [JsonPropertyName("Recommendation")]
    public string Recommendation { get; set; } = "";

    [JsonPropertyName("Final Status")]
    public string FinalStatus { get; set; } = "";

    [JsonPropertyName("Joining Date")]
    public string JoiningDate { get; set; } = "";

    [JsonPropertyName("Interviewer")]
    public string Interviewer { get; set; } = "";

    [JsonPropertyName("Resume URL")]
    public string ResumeUrl { get; set; } = "";

    [JsonPropertyName("Remarks")]
    public string Remarks { get; set; } = "";
}

